pkg/grafana/client/credential.go

```diff
@@ -0,0 +1,40 @@
+package client
```

Comment:
Your `client` package currently depends on kubernetes types and the `controller-runtime` client. If you would fetch the secrets elsewhere and just pass the values to the functions in this `client` package, you would remove the dependency and your grafana client wouldn't even know if it's being used for kubernetes things or anything else.

Reply:
That makes sense yes

Reply:
We will do it in a subsuquent PR because this means we make the main function a whole lot more complicated

Reply:
I did it mounting the secrets